api/handlers_instances.go

```diff
@@ -557,3 +557,48 @@ func (s *server) InstanceUpdateHandler(w http.ResponseWriter, r *http.Request) {
 
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func (s *server) VolumeAttachHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := s.mapAccountNumber(vars["account"])
+	id := vars["id"]
+	
+	req := &Ec2VolumeAttachment{}
```

Comment:
You are reusing this struct which is actually from the output and has some extra fields. Does it work if all the fields are not specified?

Reply:
Json decode will not have any impact because of having extra fields. However, I created a new struct `Ec2VolumeAttachmentRequest` for separating the request and response data structures, so that changing the fields for one purpose will not impact the other.